openlibrary/utils/__init__.py

```diff
@@ -127,20 +127,18 @@ def dicthash(d):
         return d
 
 
-author_olid_re = re.compile(r'^OL\d+A$')
+author_olid_embedded_re = re.compile(r'OL\d+A', re.IGNORECASE)
 
+def find_author_olid_in_string(s):
+    found = re.search(author_olid_embedded_re, s)
```

Comment:
A nice spot for three [doctests](https://docs.python.org/3/library/doctest.html) so we have proof that the function works as we expected.  This also makes sure that no one breaks the function in future code changes.
```suggestion
    """
    >>> find_author_olid_in_string("OL28641A")
    '28641'
    >>> find_author_olid_in_string("oL2864a")
    '28641'
    >>> find_author_olid_in_string("OL7500661M")
    ''
    """
    found = re.search(author_olid_embedded_re, s)
```

Reply:
plus 1! Although note the return should be `OL28641A`.